src/modules/ui/dom.js

```diff
@@ -44,6 +65,14 @@ BetterLyrics.DOM = {
     footerLink.href = sourceHref;
   },
 
+  /**
+   * Creates the footer elements including source link, Discord link, and add lyrics button.
+   *
+   * @param {string} song - Song title
+   * @param {string} artist - Artist name
+   * @param {string} album - Album name
+   * @param {number} duration - Song duration in seconds
+   */
   createFooter: function (song, artist, album, duration) {
```

Comment:
We should probably combine `createFooter` with `addFooter`. It is only called from `addFooter` and there isn't clear separation between the two functions.

May be something to save for later though

Reply:
maybe something for the typescript rewrite